rmf_site_editor/src/package_exporter/generate_package.rs

```diff
@@ -0,0 +1,236 @@
+                        "Unable to get file name from path",
+                    ))?
+                    .to_str()
+                    .ok_or(IoError::new(
+                        IoErrorKind::InvalidInput,
+                        "Unable to convert file name to str",
+                    ))?
+                    .to_owned();
+
+                let package_path =
+                    format!("{}/{}/{}", package_name, mesh_directory_name, file_name);
+                *asset_source = AssetSource::Package(package_path);
+            }
+            Result::<(), Box<dyn Error>>::Ok(())
+        })?;
+    Ok(workcell)
+}
+
+fn get_mesh_paths(workcell: &Workcell) -> Result<Vec<String>, Box<dyn Error>> {
+    let paths = workcell
+        .visuals
+        .iter()
+        .chain(workcell.collisions.iter())
+        .filter_map(|(id, visual)| get_path_if_mesh_geometry(&visual.bundle.geometry).ok()?)
+        .collect();
+    Ok(paths)
+}
+
+fn get_path_if_mesh_geometry(geometry: &Geometry) -> Result<Option<String>, Box<dyn Error>> {
+    if let Geometry::Mesh {
+        source: asset_source,
+        scale,
+    } = geometry
+    {
+        Ok(Some(get_path_to_asset_file(asset_source)?))
+    } else {
+        Ok(None)
+    }
+}
+
+fn copy_files(paths: &Vec<String>, output_directory: &Path) -> Result<(), Box<dyn Error>> {
+    for path in paths.iter() {
+        let file_name = PathBuf::from(path)
+            .file_name()
+            .ok_or(IoError::new(
+                IoErrorKind::InvalidInput,
+                "Unable to get file name from path",
+            ))?
+            .to_owned();
+        let new_path = PathBuf::from(output_directory).join(file_name);
+        match std::fs::copy(path, &new_path) {
+            Ok(_) => {}
+            Err(e) => {
+                println!(
+                    "Error copying file '{}' to '{}': {}",
+                    path,
+                    new_path.display(),
+                    e
+                );
+            }
+        }
+    }
+    Ok(())
+}
+
+fn get_path_to_asset_file(asset_source: &AssetSource) -> Result<String, Box<dyn Error>> {
+    if let AssetSource::Package(_) = asset_source {
+        let path = String::from(asset_source);
+        Ok((*urdf_rs::utils::expand_package_path(&path, None)).to_owned())
+    } else if let AssetSource::Remote(asset_name) = asset_source {
+        let mut asset_path = cache_path();
+        asset_path.push(PathBuf::from(&asset_name));
+        Ok(asset_path
+            .to_str()
+            .ok_or(IoError::new(
+                IoErrorKind::InvalidInput,
+                "Unable to convert asset_path to str",
+            ))?
+            .to_string())
+    } else if let AssetSource::Local(path) = asset_source {
+        Ok(path.clone())
+    } else {
+        Err(IoError::new(
+            IoErrorKind::Unsupported,
+            "Not a supported asset type for exporting a workcell to a package",
+        ))?
```

Comment:
This would be a bit more idiomatic written as a `match` statement rather than an `if let` chain, the unimplemented branch should probably also have all the variants written explicitly so we can make it a compile error if in the future a variant is added, i.e.:

```
match asset_source {
    AssetSource::Package(_) => [...],
    AssetSource::Remote(asset_name => [...],
    AssetSource::Local(path) => [...],
    AssetSource::Search(_) | AssetSource::OSMTile(_) | [...] => return the error
}
```

Reply:
Should be fixed in ee15eac.